src/ruby.ts

```diff
@@ -32,200 +46,199 @@ export class RubyProcess extends EventEmitter {
 	public constructor(args: LaunchRequestArguments) {
 		super();
 		this.launchArgs = args;
+		this.pendingResponses = [];
 		this.pendingCommands = [];
+		this.socketConnected = false;
 		this.state = SocketClientState.ready;
 
-		var that = this;
 		var runtimeArgs = ['--evaluation-timeout', '10'];
 		var runtimeExecutable: string;
 
 		if (process.platform === 'win32') {
 			runtimeExecutable = 'rdebug-ide.bat';
 		}
-		else if (process.platform === 'darwin') {
-			runtimeExecutable = 'rdebug-ide';
-		}
 		else {
 			// platform: linux
 			runtimeExecutable = 'rdebug-ide';
 		}
 
-		var programArgs = [];
 		var processCwd = dirname(this.launchArgs.program);
+		// `-x` only if showDebuggerOutput is set - it is also a runtime (debugger) arg, not a program arg (ie, it can't go after the filename that we're debugging)
+		if (args.showDebuggerOutput){
+			runtimeArgs.push('-x');
+		}
+		if (args.stopOnEntry){
+			runtimeArgs.push('--stop');
+		}
+		//the program is also required
+		var spawnArgs = runtimeArgs.concat(args.program, ...args.args);
+		this.debugprocess = childProcess.spawn(runtimeExecutable, spawnArgs, {cwd: processCwd});
 
-		this.debugprocess = childProcess.spawn(runtimeExecutable, [...runtimeArgs, args.program, '-xd'], {cwd: processCwd});
 		// redirect output to debug console
 		this.debugprocess.stdout.on('data', (data: Buffer) => {
-			that.emit('exeutableOutput', data);
+			this.emit('executableOutput', data);
 		});
 		this.debugprocess.stderr.on('data', (data: Buffer) => {
 			if (/^Fast Debugger/.test(data.toString())) {
-				that.debugSocketClient.connect(1234);
+				this.debugSocketClient.connect(1234);
+				if (args.showDebuggerOutput) {
+					this.emit('debuggerOutput', data);
+				}
+			}
+			else {
+				this.emit('executableStdErr', data);
 			}
-			that.emit('debuggerOutput', data)
 		});
 		this.debugprocess.on('exit', () => {
-			that.emit('debuggerProcessExit');
+			this.emit('debuggerProcessExit');
 		});
 		this.debugprocess.on('error', (error: Error) => {
-			that.emit('debuggerProcessError', error);
+			this.emit('terminalError', "Process failed: " + error.message);
 		});
 
 		this.buffer = '';
-		this.parser = new DOMParser();
+		this.parser = new DOMParser({
+			errorHandler: (type,msg)=>this.domErrorHandler(type,msg),
+			locator: domErrorLocator
+		});
 
 		this.debugSocketClient = new net.Socket( {
 			type: 'tcp4'
 		});
-
 		this.debugSocketClient.on('connect', (buffer: Buffer) => {
-			that.state = SocketClientState.connected;
-			that.emit('debuggerConnect');
+			this.state = SocketClientState.connected;
+			//first thing we have to send is the start - if stopOnEntry is
```

Comment:
Here is the catch. If you don't set any breakpoint for the source code, once you triggered  `start` command, VS Code Debug UI or our test code will never receive `initialized` event.